pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -560,20 +564,24 @@ func findClusterOperatorStatusCondition(conditions []configv1.ClusterOperatorSta
 func checkForUpgrade(cv *configv1.ClusterVersion) error {
 	results := []string{}
 	if c := findClusterOperatorStatusCondition(cv.Status.Conditions, "Invalid"); c != nil && c.Status == configv1.ConditionTrue {
-		results = append(results, fmt.Sprintf("the cluster version object is invalid, you must correct the invalid state first:\n\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  ")))
+		results = append(results, fmt.Sprintf("the cluster version object is invalid, you must correct the invalid state first:\n\n  Reason: %s\n  Message: %s", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  ")))
 	}
-	if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.OperatorDegraded); c != nil && c.Status == configv1.ConditionTrue {
-		results = append(results, fmt.Sprintf("the cluster is experiencing an upgrade-blocking error:\n\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  ")))
+	if c := findClusterOperatorStatusCondition(cv.Status.Conditions, clusterStatusFailing); c != nil && c.Status == configv1.ConditionTrue {
+		target := cv.Status.Desired.Version
+		if target == "" {
+			target = cv.Status.Desired.Image
```

Comment:
The value of cv.Status.Desired.Image might be empty too, right? But we are not checking for it.

Reply:
if `version` and `image` are both empty, we have no idea where we're going.  Seems unlikely, and also like there's not much we could do to mitigate.  What would you fall back to?

Reply:
We should check and put an appropriate error message. I do not think we can do more than that.

Reply:
I've added a fallback to `<ClusterVersion status.desired has neither a version nor image; please open a bug>` with 1900577fa -> 438e88491.